clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -59,7 +59,7 @@
 import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
-import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.PriorityBlockingQueue;
```

Comment:
Why switch data structure ?

Reply:
Due to the current policy I have set up in the KIP, this part is adapted to accomadate for OffsetCommitCallback. Currently, we split offsets being committed into two, so now we have two CommitCallbacks. If both of them are called separately, then we have two calls to the callback. We cannot let that happen to ensure that Kafka's behavior remains the same. Therefore, we need both ```OffsetCommitCompletion```s in PriorityBlockingQueue before calling OffsetCommitCallback as one call instead of two. The Hashcodes are used to compare two OffsetCommitCompletions.